jvm.c

```diff
@@ -234,13 +238,56 @@ stack_entry_t *execute(method_t *method,
             uint16_t index = ((param1 << 8) | param2);
 
             /* the method to be called */
-            method_t *own_method = find_method_from_index(index, clazz);
+            char *method_name, *method_descriptor, *class_name;
+            class_name = find_method_info_from_index(index, clazz, &method_name,
+                                                     &method_descriptor);
+            class_file_t *target_class = find_class_from_heap(class_name);
+
+            /* if class is not found, adding specific class path and finding
+             * again. This can be removed by recording path infomation in class
+             * heap
+             */
+            if (!target_class) {
+                char *tmp = malloc(strlen(class_name) + strlen(prefix) + 1);
+                strcpy(tmp, prefix);
+                strcat(tmp, class_name);
+                target_class = find_class_from_heap(tmp);
+
+                free(tmp);
+            }
+
+            /* if class is still not found, meaning that this class is not
+             * loaded in class heap, so add it to the class heap. */
+            if (!target_class) {
+                /* 7 = ".class" + 1 */
+                char *tmp = malloc((strlen(class_name) + 7 + strlen(prefix)) *
+                                   sizeof(char));
```

Comment:
In C standard, `sizeof(char)` is `1`.

See C99, section 6.5.3.4:
> When applied to an operand that has type char, unsigned char, or signed char, (or a qualified version thereof) the result is 1.

So, remove `* sizeof(char)`, which is regarded as redundant.